common/txmgr/broadcaster.go

```diff
@@ -763,12 +803,42 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) save
 	return eb.txStore.UpdateTxFatalError(ctx, etx)
 }
 
-func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) getNextSequence(address ADDR) (sequence SEQ, err error) {
-	return eb.ks.NextSequence(address, eb.chainID)
+// Used to get the next usable sequence for a transaction
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetNextSequence(address ADDR) (seq SEQ, err error) {
+	eb.sequenceLock.Lock()
+	defer eb.sequenceLock.Unlock()
+	// Get next sequence from map
+	seq, exists := eb.nextSequenceMap[address]
+	if !exists {
+		return seq, errors.New(fmt.Sprint("address not found in next sequence map: ", address))
+	}
+	return seq, nil
 }
 
-func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) incrementNextSequence(address ADDR, currentSequence SEQ, qopts ...pg.QOpt) error {
-	return eb.ks.IncrementNextSequence(address, eb.chainID, currentSequence, qopts...)
+// Used to increment the sequence in the mapping to have the next usable one available for the next transaction
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) IncrementNextSequence(address ADDR) error {
+	eb.sequenceLock.Lock()
+	defer eb.sequenceLock.Unlock()
+	// Get next sequence from map
+	seq, exists := eb.nextSequenceMap[address]
+	if !exists {
+		return errors.New(fmt.Sprint("address not found in next sequence map: ", address))
+	}
+	eb.nextSequenceMap[address] = seq.Next()
+	return nil
+}
+
+// Used to set the next sequence explicitly to a certain value
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) SetNextSequence(address ADDR, seq SEQ) error {
+	eb.sequenceLock.Lock()
+	defer eb.sequenceLock.Unlock()
+	// Get next sequence from map
+	_, exists := eb.nextSequenceMap[address]
+	if !exists {
+		return errors.New(fmt.Sprint("address not found in next sequence map: ", address))
```

Comment:
Same here. We should check existence of address in this map beforehand, or ensure that this error can't happen due to the way our logic around deleting addresses from this map works.